app/models/miq_worker/runner.rb

```diff
@@ -341,6 +341,10 @@ def do_work_loop
         @backoff = nil
       end
 
+      # Should be caught by the rescue in `#start` and will run do_exit from
+      # there.
+      raise Interrupt if @sigterm_recieved
```

Comment:
LOL, received... I noticed the spelling mistake after merging 😢 

Does it make sense to also initialize `@sigterm_received = false` somewhere?